saleor/graphql/checkout/mutations/checkout_create.py

```diff
@@ -376,7 +411,35 @@ def save(cls, info: ResolveInfo, instance: models.Checkout, cleaned_input):
                 instance.billing_address = billing_address.get_copy()
 
             instance.save()
-            create_checkout_metadata(instance)
+
+    @classmethod
+    def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
+        # Write metadata in post-save action because in "save" metadata doesn't exist in
+        # cleaned_input.
+
+        checkout_metadata = create_checkout_metadata(instance)
+
+        metadata_from_input = cleaned_input.get("metadata", [])
+        private_metadata_from_input = cleaned_input.get("private_metadata", [])
+
+        nothing_to_write = (
+            len(metadata_from_input) == 0 and len(private_metadata_from_input) == 0
```

Comment:
Nitpick: you can also do
```
if not metadata_from_input and not private_metadata_from_input:
    return
```

It's a bit easier to read.

Reply:
Okay 😄 In JS empty arrays work different 👍